beacon-chain/blockchain/process_block_helpers.go

```diff
@@ -95,11 +95,8 @@ func (s *Service) verifyBlkPreState(ctx context.Context, b interfaces.BeaconBloc
 func (s *Service) VerifyFinalizedBlkDescendant(ctx context.Context, root [32]byte) error {
 	ctx, span := trace.StartSpan(ctx, "blockChain.VerifyFinalizedBlkDescendant")
 	defer span.End()
-	finalized, err := s.store.FinalizedCheckpt()
-	if err != nil {
-		return errors.Wrap(err, "could not get finalized checkpoint")
-	}
-	fRoot := s.ensureRootNotZeros(bytesutil.ToBytes32(finalized.Root))
+	finalized := s.ForkChoicer().FinalizedCheckpoint()
+	fRoot := s.ensureRootNotZeros(finalized.Root)
```

Comment:
Unfortunately yes, checkpoints have to be zero root for genesis, so either we check against genesis root at Blockchain to zero it out for services that need that or we return zero from forkchoice. I wonder how much will break if we just return originRoot instead of zero from forkchoice